modules/src/erc1155/erc1155_base.rs

```diff
@@ -1,33 +1,26 @@
 //! Odra module implementing Erc1155 core.
+use super::events::*;
 use crate::erc1155::errors::Error;
-use crate::erc1155::events::{ApprovalForAll, TransferBatch, TransferSingle};
 use crate::erc1155::Erc1155;
-use crate::erc1155_receiver::Erc1155ReceiverRef;
-use odra::types::casper_types::bytesrepr::Bytes;
-use odra::types::OdraAddress;
-use odra::types::U256;
-use odra::{
-    contract_env::{caller, revert},
-    prelude::vec::Vec,
-    types::{event::OdraEvent, Address},
-    Mapping
-};
+use crate::erc1155_receiver::Erc1155ReceiverContractRef;
+use odra::prelude::*;
+use odra::{Address, Bytes, Mapping, Module, OdraAddress, U256};
 
 /// The ERC1155 base implementation.
 #[odra::module(events = [ApprovalForAll, TransferBatch, TransferSingle])]
 pub struct Erc1155Base {
-    pub balances: Mapping<Address, Mapping<U256, U256>>,
-    pub approvals: Mapping<Address, Mapping<Address, bool>>
+    pub balances: Mapping<(Address, U256), U256>,
+    pub approvals: Mapping<(Address, Address), bool>
 }
 
 impl Erc1155 for Erc1155Base {
     fn balance_of(&self, owner: &Address, id: &U256) -> U256 {
-        self.balances.get_instance(owner).get_or_default(id)
+        self.balances.get_or_default(&(*owner, *id))
     }
 
-    fn balance_of_batch(&self, owners: &[Address], ids: &[U256]) -> Vec<U256> {
+    fn balance_of_batch(&self, owners: Vec<Address>, ids: Vec<U256>) -> Vec<U256> {
```

Comment:
Why did you change a slice to a vec? Does it work with slices?

Reply:
It did not in some place, so I changed it everywhere